src/saving/serializers/EntityReferenceConverter.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Reflection;
+using Newtonsoft.Json;
+
+public class EntityReferenceConverter : JsonConverter
+{
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+    {
+        if (value == null)
+        {
+            writer.WriteNull();
+            return;
+        }
+
+        var objectType = value.GetType();
+
+        var genericTypes = objectType.GenericTypeArguments;
+
+        if (genericTypes.Length != 1)
+            throw new JsonException("Invalid generic types for EntityReference");
```

Comment:
Could we print out the objectType variable as well? I imagine that could have handy information.

Reply:
Maybe? But this should *only* happen when someone is working on new code and they mess this up. Meaning that it's probably pretty easy to tell what's wrong. If not, they can turn off save error catching and run their debugger. This is pretty similar to the other JSON errors that aren't about input, but straight up just mistakes in the code.